Encountered a InternalServerError from host runtime when deploy

### Plugin name and version
azure-functions-maven-plugin 1.7.0
### Plugin configuration in your `pom.xml`
```
<configuration>
 <appName>${functionAppName}</appName>
 <resourceGroup>harryfunc06</resourceGroup>
 <appServicePlanName>harryfunc06-app-service-plan</appServicePlanName>
 <region>eastus2</region>
 <pricingTier>EP1</pricingTier>
 <runtime>
 <os>docker</os>
 <image>xxxxxx/func-jdk11:0.1.0</image>
 <serverId>SERVERID1</serverId>
 </runtime>
 <appSettings>
 <property>
 <name>FUNCTIONS_EXTENSION_VERSION</name>
 <value>~3</value>
 </property>
 <property>
 <name>AzureWebJobsStorage</name>
 <value>xxxxx</value>
 </property>
 <property>
 <name>connectionbus</name>
 <value>xxxxxxxx</value>
 </property>
 <property>
 <name>xxxxxxxx</value>
 </property>
 <property>
 <name>connectionhub</name>
 <value>xxxxxxxxx</value>
 </property>
 </appSettings>
 </configuration>
```
### Actual behavior

```
[INFO] The specified function app does not exist. Creating a new function app...
[INFO] Set function worker runtime to java.
[INFO] Creating application insights...
[INFO] Successfully created the application insights harryfunc04 for this Function App. You can visit https://portal.azure.com/#resource/subscriptions/xxxxx/resourceGroups/harryfunc06/providers/microsoft.insights/components/harryfunc06/overview to view your Application Insights component.
[INFO] Successfully created the function app: harryfunc06.
[INFO] Trying to deploy the function app...
[INFO] Skip deployment for docker functions
[INFO] Successfully deployed the function app at https://harryfunc06.azurewebsites.net.
[INFO] Syncing triggers and fetching function information (Attempt 1/3)...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 03:03 min
[INFO] Finished at: 2020-08-11T15:48:49+08:00
[INFO] Final Memory: 48M/121M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.microsoft.azure:azure-functions-maven-plugin:1.7.0:deploy (default-cli) on project harryfunc02: Status code 400, {"Code":"BadRequest","Message":"Encountered an error (InternalServerError) from host runtime.","Target":null,"Details":[{"Message":"Encountered an error (InternalServerError) from host runtime."},{"Code":"BadRequest"},{"ErrorEntity":{"Code":"BadRequest","Message":"Encountered an error (InternalServerError) from host runtime."}}],"Innererror":null} -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java
+++ b/azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java
@@ -359,14 +359,19 @@ public class DeployMojo extends AbstractFunctionMojo {
         for (int i = 0; i < LIST_TRIGGERS_MAX_RETRY; i++) {
             Thread.sleep(LIST_TRIGGERS_RETRY_PERIOD_IN_SECONDS * 1000);
             Log.info(String.format(SYNCING_TRIGGERS_AND_FETCH_FUNCTION_INFORMATION, i + 1, LIST_TRIGGERS_MAX_RETRY));
-            functionApp.syncTriggers();
-            final List<FunctionResource> triggers = getAzureClient().appServices().functionApps()
-                    .listFunctions(getResourceGroup(), getAppName()).stream()
-                    .map(envelope -> FunctionResource.parseFunction(envelope))
-                    .filter(function -> function != null)
-                    .collect(Collectors.toList());
-            if (CollectionUtils.isNotEmpty(triggers)) {
-                return triggers;
+            try {
+                functionApp.syncTriggers();
+                final List<FunctionResource> triggers = getAzureClient().appServices().functionApps()
+                        .listFunctions(getResourceGroup(), getAppName()).stream()
+                        .map(envelope -> FunctionResource.parseFunction(envelope))
+                        .filter(function -> function != null)
+                        .collect(Collectors.toList());
+                if (CollectionUtils.isNotEmpty(triggers)) {
+                    return triggers;
+                }
+            } catch (RuntimeException e) {
+                // swallow service exception while list triggers
+                continue;
             }
         }
         throw new AzureExecutionException(NO_TRIGGERS_FOUNDED);

